temporal-sdk/src/test/java/io/temporal/common/converter/ProtoPayloadConverterTest.java

```diff
@@ -96,6 +101,42 @@ public void testCustomProto() {
     assertEquals(execution, converted);
   }
 
+  @Test
+  public void testProtoMessageType() {
+    DataConverter converter = DataConverter.getDefaultInstance();
+    WorkflowExecution execution =
+        WorkflowExecution.newBuilder()
+            .setWorkflowId(UUID.randomUUID().toString())
+            .setRunId(UUID.randomUUID().toString())
+            .build();
+    Optional<Payloads> data = converter.toPayloads(execution);
+    Payloads payloads = data.get();
+    Object field = payloads.getField(payloads.getDescriptorForType().findFieldByName("payloads"));
+    Payload payload = (Payload) ((List<?>) field).get(0);
+    Object metadata = payload.getField(payload.getDescriptorForType().findFieldByName("metadata"));
+    MapEntry<?, ?> secondMetadata = (MapEntry<?, ?>) ((List<?>) metadata).get(1);
+    assertEquals("messageType", secondMetadata.getKey());
+    assertEquals(
+        "temporal.api.common.v1.WorkflowExecution",
+        ((ByteString) secondMetadata.getValue()).toString(StandardCharsets.UTF_8));
+  }
+
+  @Test
+  public void testProtoMessageTypeExclusion() {
+    DataConverter converter = DefaultDataConverter.newDefaultInstance(true);
+    WorkflowExecution execution =
+        WorkflowExecution.newBuilder()
+            .setWorkflowId(UUID.randomUUID().toString())
+            .setRunId(UUID.randomUUID().toString())
+            .build();
+    Optional<Payloads> data = converter.toPayloads(execution);
+    Payloads payloads = data.get();
+    Object field = payloads.getField(payloads.getDescriptorForType().findFieldByName("payloads"));
+    Payload payload = (Payload) ((List<?>) field).get(0);
+    Object metadata = payload.getField(payload.getDescriptorForType().findFieldByName("metadata"));
+    assertEquals(1, ((List<?>) metadata).size());
+  }
+
```

Comment:
I could also test the warning message if there's a way to mock/spy on `log.warn`?

Reply:
No need really, this is waaaay too deep.